ui/src/overview.tsx

```diff
@@ -359,83 +429,132 @@ function DemographicCharts({ cohort }: DemographicChartsProps) {
+              primaryPropertyComponents.push(`${dataBoolVal}`);
+            }
+          }
+        });
+
+        // Add the primary property grouping to the chart data if it hasn't already
+        const primaryPropertyString = primaryPropertyComponents.join(" ");
+        if (!chartsData[index].has(primaryPropertyString)) {
+          if (config.stackedProperty) {
+            chartsData[index].set(primaryPropertyString, new Map());
+          } else {
+            chartsData[index].set(primaryPropertyString, 0);
+          }
+        }
 
-      // Configure set of races returned in results
-      if (!demographicRaces.has(race)) {
-        demographicRaces.add(race);
-      }
+        // Determine the counts for the chart's stacked property if it exists
+        const stackedProperty = config.stackedProperty;
+        if (stackedProperty) {
+          const stackedPropertyValues = Object.entries(
+            demographicData[i].definition ?? {}
+          ).find((data) => data[0] === stackedProperty.key);
+
+          if (stackedPropertyValues) {
+            const stackedIntegerVal = stackedPropertyValues[1].int64Val;
+            const stackedStringVal = stackedPropertyValues[1].stringVal;
+            const stackedBoolVal = stackedPropertyValues[1].boolVal;
+            let stackValue;
+            if (stackedIntegerVal) {
+              stackValue = stackedIntegerVal.toString();
+            } else if (stackedStringVal) {
+              stackValue = stackedStringVal;
+            } else if (stackedBoolVal) {
+              stackValue = stackedBoolVal.toString();
+            }
+
+            if (stackValue) {
+              if (
+                !chartsData[index].get(primaryPropertyString).has(stackValue)
+              ) {
+                chartsData[index].get(primaryPropertyString).set(stackValue, 0);
+              }
+
+              const prevCount = chartsData[index]
+                .get(primaryPropertyString)
+                .get(stackValue);
+              chartsData[index]
+                .get(primaryPropertyString)
+                .set(stackValue, prevCount + count);
+            }
+
+            stackedProperties[index].add(stackValue);
+          }
+        } else {
+          const prevCount = chartsData[index].get(primaryPropertyString);
+          chartsData[index].set(primaryPropertyString, prevCount + count);
+        }
+      });
     }
 
     return {
       totalCount,
-      demographicsByGender: Array.from(demographicsByGender, ([key, value]) => {
-        return {
-          name: key,
-          count: value,
-          color: barColours[0],
-        };
-      }),
-      demographicRaces: Array.from(demographicRaces),
-      demographicsByGenderAgeRace: Array.from(
-        demographicsByGenderAgeRace,
-        ([key, value]) => {
-          return {
-            ageRange: key,
-            ...Object.fromEntries(value),
-          };
+      chartsData: chartsData.map((chart, index) => {
```

Comment:
The chart data needs to be sorted (alphabetically is fine) so that the bars appear in a consistent order.